src/main.py

```diff
@@ -19,22 +25,13 @@
 app = Flask(__name__)
 
 
-def handle_event(data, headers):
-    if headers.get("X-GitHub-Event") != "workflow_job":
-        # We return 200 to make webhook not turn red
-        return {"reason": "Event not supported."}, 200
-
-    if data["action"] != "completed":
-        return ({"reason": "We cannot do anything with this workflow state."}, 200)
-
-    send_trace(data["workflow_job"])
-    return {"reason": "OK"}, 200
-
-
 @app.route("/", methods=["POST"])
 def main():
     payload = {"reason": "There was an error."}
     http_code = 500
-    payload, http_code = handle_event(request.json, request.headers)
-
-    return jsonify(payload), http_code
+    # Top-level crash preventing try block
+    try:
+        reason, http_code = handle_event(request.json, request.headers)
+        payload = {"reason": reason}
+    finally:
+        return jsonify(payload), http_code
```

Comment:
`try finally: return` is almost always a bug -- I believe it even raises a `SyntaxWarning` in newer python versions

I would prefer to see explicitly:

```python
try:
    ...
except Exception:  # though ideally it would catch a more specific exception and log it somewhere
    return jsonify({"reason": "There was an error."}), 500
```